rmw_fastrtps_shared_cpp/src/rmw_wait.cpp

```diff
@@ -114,89 +55,92 @@ __rmw_wait(
   // error.
   // - Heap is corrupt.
   // In all three cases, it's better if this crashes soon enough.
-  auto wait_set_info = static_cast<CustomWaitsetInfo *>(wait_set->data);
-  std::mutex * conditionMutex = &wait_set_info->condition_mutex;
-  std::condition_variable * conditionVariable = &wait_set_info->condition;
+  auto fastdds_wait_set = static_cast<eprosima::fastdds::dds::WaitSet *>(wait_set->data);
+  bool skip_wait = false;
```

Comment:
is doing this really a good optimization?

Reply:
If I remember well something similar was being done before and we decide to keep it.

Reply:
ok sounds good.
I don't think that's helping much, but no need to remove that now.